beacon_node/client/src/notifier.rs

```diff
@@ -213,6 +184,102 @@ pub fn spawn_notifier<T: BeaconChainTypes>(
     Ok(())
 }
 
+fn eth1_logging<T: BeaconChainTypes>(beacon_chain: &BeaconChain<T>, log: &Logger) {
+    let current_slot = beacon_chain
+        .slot()
+        .unwrap_or(beacon_chain.spec.genesis_slot);
+
+    if let Ok(head_info) = beacon_chain.head_info() {
+        // Perform some logging about the eth1 chain
+        if let Some(eth1_chain) = beacon_chain.eth1_chain.as_ref() {
+            if let Some(status) =
+                eth1_chain.sync_status(head_info.genesis_time, current_slot, &beacon_chain.spec)
+            {
+                debug!(
+                    log,
+                    "Eth1 cache sync status";
+                    "eth1_head_block" => status.head_block_number,
+                    "latest_cached_block_number" => status.latest_cached_block_number,
+                    "latest_cached_timestamp" => status.latest_cached_block_timestamp,
+                    "voting_target_timestamp" => status.voting_target_timestamp,
+                    "ready" => status.lighthouse_is_cached_and_ready
+                );
+
+                if !status.lighthouse_is_cached_and_ready {
```

Comment:
This status is always `false` before genesis, because the place where it's calculated uses the start of the genesis voting period:

https://github.com/sigp/lighthouse/blob/f7183098ee799fcc091db416d28d389c03074cc7/beacon_node/beacon_chain/src/eth1_chain.rs#L79-L91

https://github.com/sigp/lighthouse/blob/f7183098ee799fcc091db416d28d389c03074cc7/beacon_node/beacon_chain/src/eth1_chain.rs#L127-L128

Combined with the missing follow distance in the calculation of `voting_target_timestamp` function, I think this causes nodes to _always_ appear out of sync before genesis, with an apparent non-zero distance to head.

I think a good fix would be to move the "virtual voting period" logic into the `get_sync_status` function, so that it can calculate an accurate `lighthouse_is_cached_and_ready` based on the current time, etc.